### PR TITLE
feat: Ability to collapse/expand table to simplify the visualization

### PR DESCRIPTION
**Pull request by tv-long**
@BOCOVO 
Please check my changes according to the issue #49

### CODE REVIEW

**Comment by BOCOVO** [comment]
> @BOCOVO Please check my changes according to the issue #49

Hello @tv-long . 
I'm very happy to see this PR.
Thank you very much for working on the feature.

I'll take a look at it as soon as possible, and then we'll release that 🚀

File: packages/json-table-schema-visualizer/src/providers/TableDetailLevelProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+interface TabelLevelDetailProviderProps {
+  children: ReactNode;
+  level?: TableDetailLevel;
+}
+
+const TabelLevelDetailProvider = ({
+  children,
+}: TabelLevelDetailProviderProps) => {
+  const [state, setState] = useState(detailLevelStore.getCurrentDetailLevel());
```
Comment (BOCOVO) [inline packages/json-table-schema-visualizer/src/providers/TableDetailLevelProvider.tsx:28]: A title suggestion
```suggestion
  const [state, setState] = useState( () => detailLevelStore.getCurrentDetailLevel() );
```
The advantage is that the `detailLevelStore.getCurrentDetailLevel` will be called only once. At the first rendering React will call it to get the initial value. For the subsequent rendering, React will no longer call it.

File: packages/json-table-schema-visualizer/src/types/tabelDetailLevel.ts
Comment (BOCOVO) [inline packages/json-table-schema-visualizer/src/types/tabelDetailLevel.ts:1]: I think that you meant `tableDetailLevel`.
> [!NOTE]
> I'm talking about the file name.

**Review by BOCOVO — CHANGES_REQUESTED** [review]
I've tested it, and it works well. However, I have left a small suggestion for change, mainly in the text.

**Comment by tv-long** [comment]
@BOCOVO 
I made changes according to your feedback
Please check

**Comment by BOCOVO** [comment]
Thank you very much @tv-long.

I'm going to merge these, and we'll publish them this very evening 🔥 .

**Review by BOCOVO — APPROVED** [review]

**Comment by BOCOVO** [comment]
close: #49

**Pull request merged**